pageserver/src/tenant/storage_layer.rs

```diff
@@ -431,11 +431,11 @@ pub trait PersistentLayer: Layer {
     fn local_path(&self) -> Option<PathBuf>;
 
     /// Iterate through all keys and values stored in the layer
-    fn iter(&self, ctx: &RequestContext) -> Result<LayerIter<'_>>;
+    fn iter(self: Arc<Self>, ctx: &RequestContext) -> Result<LayerIter<'static>>;
```

Comment:
I thought the [original PR](https://github.com/neondatabase/neon/pull/4265/) from me of adding spawn_blocking is rather simple compared with this patch? Is it the lifetime thing that we have to change `iter` to `Arc<Self>`?

Reply:
Yeah, see the individual commits.

As I said, 

> However, I'm wondering whether we should just use (1) and alert if lock acquisition takes too long.